src/test/java/org/apache/commons/lang3/time/CalendarUtilsTest.java

```diff
@@ -128,5 +131,13 @@ public void testToZonedDateTime(final String id) {
         final ZonedDateTime zdt1 = ZonedDateTime.of(1, 2, 3, 4, 5, 6, 0, zoneId);
         calendar.setTimeInMillis(zdt1.toInstant().toEpochMilli());
         assertEquals(ZonedDateTime.ofInstant(zdt1.toInstant(), calendar.getTimeZone().toZoneId()), new CalendarUtils(calendar).toZonedDateTime());
+
+    @Test
```

Comment:
Hello @asgh,

Thank you for your reply and update.

I don't think the implementation is correct because it does not take into account the Calendar's time zone.

You should be able to see this if you test with time zones and adjust the test as in my example test using:

```java
    @ParameterizedTest
    @MethodSource(TimeZonesTest.TIME_ZONE_GET_AVAILABLE_IDS)
```

See also the other tests in this class.

TY!

Reply:
@garydgregory 

I'm really not following you. For this to be wrong this.getDayOfMonth() would have to be wrong. Are you saying that function returns the wrong day of the month?

Also your Assert is comparing an object with a Time to one without a Time - obviously they are going to be different!

And the entire point is that LocalDate doesn't *have* a timezone. I'm only setting the timezone in the test so that it correctly converts the UnixTime into a Month and Day (otherwise it could vary +- a day depending on what timezone you pick).

If you look at the actual function in the code there is no timezone there are all.

Reply:
Hello @asgh 

One of the reasons for confusion perhaps is that the test data does not look right.

How did you get from `-27078001200000` to `LocalDate.of(1111, Month.DECEMBER, 1)` when:
```
    public void testToLocalDate() {
        final Calendar calendar = new GregorianCalendar(TimeZone.getTimeZone(TimeZones.GMT_ID));
        calendar.setTimeInMillis(-27078001200000L);
        assertEquals("1111-12-08T05:00:00Z", calendar.toInstant().toString());
        assertEquals(LocalDate.of(1111, Month.DECEMBER, 8), new CalendarUtils(calendar).toLocalDate());
        calendar.setTimeInMillis(1614700215000L);
        assertEquals(LocalDate.of(2021, Month.MARCH, 2), new CalendarUtils(calendar).toLocalDate());
    }
```
?
See [`CalendarUtilsTest.testToLocalDate()`](https://github.com/apache/commons-lang/blob/3a8293af2b8cdab191a0e7ca6e306e0a2ac1e9e8/src/test/java/org/apache/commons/lang3/time/CalendarUtilsTest.java#L100-L107).